test/e2e/alertmanager_test.go

```diff
@@ -291,3 +285,186 @@ receivers:
+	defer ctx.Cleanup(t)
+	ns := ctx.CreateNamespace(t, framework.KubeClient)
+
+	whReplicas := int32(1)
+	whdpl := &v1beta1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "alertmanager-webhook",
+		},
+		Spec: v1beta1.DeploymentSpec{
+			Replicas: &whReplicas,
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": "alertmanager-webhook",
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "webhook-server",
+							Image: "quay.io/brancz/prometheus-alertmanager-test-webhook",
+							Ports: []v1.ContainerPort{
+								{
+									Name:          "web",
+									ContainerPort: 5001,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	whsvc := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "alertmanager-webhook",
+		},
+		Spec: v1.ServiceSpec{
+			Type: v1.ServiceTypeClusterIP,
+			Ports: []v1.ServicePort{
+				v1.ServicePort{
+					Name:       "web",
+					Port:       5001,
+					TargetPort: intstr.FromString("web"),
+				},
+			},
+			Selector: map[string]string{
+				"app": "alertmanager-webhook",
+			},
+		},
+	}
+	if _, err := framework.KubeClient.CoreV1().Services(ns).Create(whsvc); err != nil {
+		t.Fatal(err)
+	}
+	if _, err := framework.KubeClient.ExtensionsV1beta1().Deployments(ns).Create(whdpl); err != nil {
+		t.Fatal(err)
+	}
+	err := testFramework.WaitForPodsReady(framework.KubeClient, ns, 5*time.Minute, 1,
+		metav1.ListOptions{
+			LabelSelector: fields.SelectorFromSet(fields.Set(map[string]string{
+				"app": "alertmanager-webhook",
+			})).String(),
+		},
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	alertmanager := framework.MakeBasicAlertmanager("rolling-deploy", 2)
+	alertmanager.Spec.Version = "v0.7.0"
+	amsvc := framework.MakeAlertmanagerService(alertmanager.Name, "test", v1.ServiceTypeClusterIP)
+	amcfg := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("alertmanager-%s", alertmanager.Name),
+		},
+		Data: map[string][]byte{
+			"alertmanager.yaml": []byte(fmt.Sprintf(`
+global:
+  resolve_timeout: 5m
+
+route:
+  group_by: ['alertname']
+  group_wait: 10s
+  group_interval: 10s
+  repeat_interval: 1h
+  receiver: 'webhook'
+receivers:
+- name: 'webhook'
+  webhook_configs:
+  - url: 'http://%s.%s.svc:5001/'
+inhibit_rules:
+  - source_match:
+      severity: 'critical'
+    target_match:
+      severity: 'warning'
+    equal: ['alertname', 'dev', 'instance']
+`, whsvc.Name, ns)),
+		},
+	}
+
+	if _, err := framework.KubeClient.CoreV1().Secrets(ns).Create(amcfg); err != nil {
+		t.Fatal(err)
+	}
+	if _, err := framework.MonClient.Alertmanagers(ns).Create(alertmanager); err != nil {
+		t.Fatal(err)
+	}
+	if _, err := testFramework.CreateServiceAndWaitUntilReady(framework.KubeClient, ns, amsvc); err != nil {
+		t.Fatal(err)
+	}
+
+	p := framework.MakeBasicPrometheus(ns, "test", "test", 3)
+	framework.AddAlertingToPrometheus(p, ns, alertmanager.Name)
```

Comment:
Instead of `alertmanager.Name`, shouldn't this be `amsvc.Name`? I know in the end it is the same string, but just in case we change the Alertmanager name, but not the Alertmanager service.